### PR TITLE
fix: Registration occurs with a password that does not meet the requirements to Quince

### PR DESCRIPTION
**Pull request by DmytroAlipov**
### Description
The behavior of the registration form is very different from the behavior of this form for the Palm release.
Even with incorrectly filled fields, the form can still be submitted. In certain instances, registration may succeed, such as when the password consists of merely two characters.

There’s an issue where the error message doesn’t disappear even after the error is corrected and the focus is shifted to another field. Here’s how to reproduce it:

1. Fill in all fields correctly.
2. Enter a username or email with a space.
3. Submit a request to create an account.
4. An error will appear in the field where there is a space.
5. The cursor should already be in the field with the error (so there’s no need to click on this field again). Remove the space.
6. Move the cursor to another field.
7. The error message remains for the field where the space was initially present, even though it has been corrected.

**My proposals are aimed at returning the behavior as happened on Palm.**

**PS:** I’ve noticed that you’ve been conducting numerous experiments on the master branch. To avoid any conflicts, I’ve refrained from making these changes directly to the master. If there are any upcoming plans for the registration page, please consider these issues before the release of Redwood.

#### Screenshots/sandbox (optional):

- Before

screen_3

after clicking on "Create an account for free" button (the request to the backend is still sent even if there is a space):
screen_4

- After

prevent a request to the backend if the password does not meet the conditions:
screen_1

screen_2

#### Post-merge Checklist

* [ ] Deploy the changes to prod after verifying on stage or ask **@openedx/2u-vanguards** to do it. 
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @DmytroAlipov! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by arbrandes** [comment]
@DmytroAlipov, nice catch, thanks!  Can you submit a PR to master first, if it doesn't already exist?  If it does, can you point to it here?

**Comment by DmytroAlipov** [comment]
> @DmytroAlipov, nice catch, thanks! Can you submit a PR to master first, if it doesn't already exist? If it does, can you point to it here?

done

**Comment by DmytroAlipov** [comment]
@arbrandes @syedsajjadkazmii Everything is done for both Master and Quince.

**Comment by arbrandes** [comment]
Great!  Let's wait to merge this until the master PR lands, though.

**Comment by syedsajjadkazmii** [comment]
Hi @DmytroAlipov, master PR have been merged. Please look at this [comment](https://github.com/openedx/frontend-app-authn/pull/1184#discussion_r1522554055) and make that change in this branch too.

**Comment by DmytroAlipov** [comment]
Hi @syedsajjadkazmii 
If we are talking about this, then I already did this when it was changed in the master:
```
if (registrationError[name]) {
  dispatch(clearRegistrationBackendError(name));
}
```

File: src/register/data/utils.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { getConfig, snakeCaseObject } from '@edx/frontend-platform';
 
-import { LETTER_REGEX, NUMBER_REGEX } from '../../data/constants';
+import { LETTER_REGEX, NUMBER_REGEX, VALID_EMAIL_REGEX } from '../../data/constants';
 import messages from '../messages';
+import validateEmail from '../RegistrationFields/EmailField/validator';
+import validateName from '../RegistrationFields/NameField/validator';
+import validateUsername from '../RegistrationFields/UsernameField/validator';
+
+export const emailRegex = new RegExp(VALID_EMAIL_REGEX, 'i');
```
Comment (syedsajjadkazmii) [inline src/register/data/utils.js:9]: this can be removed

**Review by syedsajjadkazmii — APPROVED** [review]

File: src/register/data/utils.js (hunk 1 above)
Comment (DmytroAlipov) [reply src/register/data/utils.js]: Hi @syedsajjadkazmii 
Why remove `validateUsername` and `validateName`? Or was this noted by chance?

Comment (syedsajjadkazmii) [reply src/register/data/utils.js]: > Hi @syedsajjadkazmii Why remove `validateUsername` and `validateName`? Or was this noted by chance?

I only meant for emailRegex. Github picked those lines as well. :)

**Comment by openedx-webhooks** [comment]
@DmytroAlipov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-authn&entry.867055334=https://github.com/openedx/frontend-app-authn/pull/1182&entry.1484655318=https://github.com/DmytroAlipov&entry.752974735=2024-02-29+16:33&entry.1917517419=2024-03-18+07:56&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**